src/components/LoanCards/KivaClassicBasicLoanCard.vue

```diff
@@ -511,6 +471,22 @@ export default {
 	beforeDestroy() {
 		this.destroyViewportObserver();
 	},
+	created() {
+		try {
+			// Attempt to read the loan card fragment from the cache
+			// If cache is missing fragment fields, this will throw and invariant error
+			const loanCardFieldFragment = this.apollo.readFragment({
+				id: `LoanPartner:${this.loanId}`,
```

Comment:
Should this be LoanBasic to cover both LoanDirect and LoanPartner?

Reply:
From what I see in the Apollo Cache this will be either `LoanPartner` or `LoanDirect`...so maybe we need to somehow switch on the distribution type...

Reply:
I think it works to use `LoanBasic` there, as all the fields being requested exist on that type. The Apollo client understands the relationship between `LoanBasic`, `LoanPartner`, and `LoanDirect`, because we define the [possible types](https://github.com/kiva/ui/blob/5834fb39246bf87a97582e608fbbd008269afa2b/src/api/apollo.js#L29).

Reply:
That would be great!

Reply:
Using `LoanBasic` Does appear to work in the VM!

Reply:
hmm that's curious and the change I also made... I was testing by using the carousel switcher on the home page and seeing that clicking a previously loaded category would no longer hit the server so I thought it was getting it from the cache...

Reply:
Which context were you using?

Reply:
I tried both the homepage multi carousel and the lbc eco page -- attempting to read all 3 different fragments LoanBasic, LoanDirect and LoanPartner, but the only one that I can get to return anything is LoanPartner

Reply:
> I think it works to use `LoanBasic` there, as all the fields being requested exist on that type. The Apollo client understands the relationship between `LoanBasic`, `LoanPartner`, and `LoanDirect`, because we define the [possible types](https://github.com/kiva/ui/blob/5834fb39246bf87a97582e608fbbd008269afa2b/src/api/apollo.js#L29).

@emuvente thoughts on this? I could not get LoanBasic to work. And if I look at my cache I only see LoanPartner or LoanDirect entries